odk/pom.xml

```diff
@@ -0,0 +1,127 @@
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <parent>
+        <artifactId>modules</artifactId>
+        <groupId>org.motechproject</groupId>
+        <version>1.0-SNAPSHOT</version>
+        <relativePath>../</relativePath>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>odk</artifactId>
+    <description>ODK Module</description>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>bundle</packaging>
+    <name>ODK Module</name>
+
+    <properties>
+        <modules.root.dir>${basedir}/..</modules.root.dir>
+    </properties>
+
+    <dependencies>
+
+        <dependency>
+            <groupId>${motech.groupId}</groupId>
+            <artifactId>motech-platform-web-security</artifactId>
+            <version>${motech.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient-osgi</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpcore-osgi</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>${motech.groupId}</groupId>
+            <artifactId>motech-tasks</artifactId>
+            <version>${motech.version}</version>
+            <optional>true</optional>
+        </dependency>
+
+
+
+        <dependency>
+            <groupId>${motech.groupId}</groupId>
+            <artifactId>motech-admin</artifactId>
+            <version>${motech.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-test-mvc</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+
+    </dependencies>
+
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Blueprint-Enabled>true</Blueprint-Enabled>
+                        <Context-Path>odk</Context-Path>
+                        <Resource-Path>odk/resources</Resource-Path>
+                        <Import-Package>
+                            net.sf.cglib.core,
+                            net.sf.cglib.proxy,
+                            net.sf.cglib.reflect,
+                            org.aopalliance.aop,
+                            org.motechproject.admin.domain,
+                            org.motechproject.admin.messages,
+                            org.motechproject.config.service,
+                            *
+                        </Import-Package>
+                        <Export-Package>
+                            org.motechproject.odk.service;version=${project.version},

```

Comment:
Not really, anything that is used in the integration tests should get exported. The idea is that the integration tests create a client bundle that uses our bundle and tests it API, simulating how it would get used in the real world. This integration testing can bypass testing internal classes, since it tests them indirectly when testing the API. The design of internal classes can be tested in unit tests. If you are using some of these classes in the integration tests, perhaps there is a case to make for keeping them exported?